PlagiarismPlugin.inc.php

```diff
@@ -37,25 +37,67 @@ public function getDisplayName() {
 	 * @copydoc Plugin::getDescription()
 	 */
 	public function getDescription() {
-		return Config::getVar('ithenticate', 'ithenticate')?__('plugins.generic.plagiarism.description'):__('plugins.generic.plagiarism.description.seeReadme');
+		return __('plugins.generic.plagiarism.description');
 	}
 
 	/**
 	 * @copydoc LazyLoadPlugin::getCanEnable()
 	 */
-	function getCanEnable() {
-		if (!parent::getCanEnable()) return false;
-		return Config::getVar('ithenticate', 'ithenticate');
+	function getCanEnable($contextId = null) {
+		return !Config::getVar('ithenticate', 'ithenticate');
+	}
+
+	/**
+	 * @copydoc LazyLoadPlugin::getCanDisable()
+	 */
+	function getCanDisable($contextId = null) {
+		return !Config::getVar('ithenticate', 'ithenticate');
 	}
 
 	/**
 	 * @copydoc LazyLoadPlugin::getEnabled()
 	 */
 	function getEnabled($contextId = null) {
-		if (!parent::getEnabled($contextId)) return false;
-		return Config::getVar('ithenticate', 'ithenticate');
+		return parent::getEnabled($contextId) || Config::getVar('ithenticate', 'ithenticate');
+	}
+
+	/**
+	 * Fetch credentials from config.inc.php, if available
+	 * @return array username and password, or null(s)
+	**/
+	function getForcedCredentials() {
+		$request = Application::getRequest();
+		$context = $request->getContext();
+		$contextPath = $context->getPath();
+		$username = Config::getVar('ithenticate', 'username[' . $contextPath . ']',
+				Config::getVar('ithenticate', 'username'));
+		$password = Config::getVar('ithenticate', 'password[' . $contextPath . ']',
+				Config::getVar('ithenticate', 'password'));
+		return array($username, $password);
 	}
 
+	/**
+	 * Send the editor an error message
+	 * @param $submissionid int
+	 * @param $message string
+	 * @return void
+	**/
+	public function sendErrorMessage($submissionid, $message) {
+		$request = Application::getRequest();
+		$context = $request->getContext();
+		import('classes.notification.NotificationManager');
+		$notificationManager = new NotificationManager();
+		$roleDao = DAORegistry::getDAO('RoleDAO'); /* @var $roleDao RoleDAO */
+		// Get the managers.
+		$managers = $roleDao->getUsersByRoleId(ROLE_ID_MANAGER, $context->getId());
+		$managersArray = $managers->toAssociativeArray();
+		$allUserIds = array_keys($managersArray);
+		foreach ($allUserIds as $userId) {
+			$notificationManager->createTrivialNotification($userId, NOTIFICATION_TYPE_ERROR, array('contents' => __('plugins.generic.plagiarism.errorMessage', array('submissionId' => $submissionid, 'errorMessage' => $message))));
```

Comment:
Agreed.  We'd like to migrate these to Tasks instead of Trivial Notifications, but we didn't come up with an example of creating a Task in the Sprint.  Error messages should be routed to Journal Managers or Administrators, depending on if the settings come from the form, or from config.inc.php.  I could also imagine creating Tasks for Editors in the future as part of the review of the plagiarism check.  I'd prefer to take care of this in an enhancement round after an initial merge.